[ Milestone 6 ] Preserve data

Harrynoqs opened the pull request:
- On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)
- On startup of your app load all data from files.

CakeInTech committed:
- generate a people.json file

CakeInTech committed:
- commit on changes

CakeInTech committed:
- Make changes to the methods for load

CakeInTech committed:
- Make changes to the book class

Harrynoqs committed:
Make changes to rental.rb and person.rb

Harrynoqs committed:
Removed linters errors

ShadyShawkat commented on app.rb:
```diff
@@ -12,11 +15,13 @@ def initialize
     @books = []
     @people = []
     @rentals = []
+    load_data
```
- [x] Please make sure that the app can function properly with no errors shown if there are one or more missing files, as per [this](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_preserve_data.md#:~:text=Remember%20to%20handle%20the%20case%20when%20one%20or%20more%20of%20the%20files%20are%20missing) requirement.

![image](https://user-images.githubusercontent.com/62290545/212199646-475eb406-da08-4e84-a9c6-7a3bf6482c75.png)

CakeInTech replied:
- Created a condition for when the json file doesn't exist. 😄

ShadyShawkat commented on app.rb:
```diff
@@ -25,126 +30,163 @@ def list_all_books
+    end
+
+    rentals_data = JSON.parse(File.read('rentals.json'))
+    rentals_data.each do |rental|
+      date = rental['date']
+      person = @people.find { |p| p.id == rental['person_id'] }
+      book = @books.find { |b| b.id == rental['book_id'] }
+      @rentals << Rental.new(date, person, book)
+    end
```
- [x] Kindly make sure that the rentals are preserved correctly after the app is reopened. Currently, when I try to save a rental and show it after I reopen the app, I get the following result:

![image](https://user-images.githubusercontent.com/62290545/212200572-b7908f62-1153-4172-82df-4c2062b8907b.png)

CakeInTech replied:
- It's fixed now rentals is preserved correctly when reopened and you can get the saved person id by listing all people. 😄

ShadyShawkat reviewed (changes requested):
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

![Image](https://media.giphy.com/media/UdaSUYClr92cjWpeVx/giphy.gif)

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

CakeInTech committed:
- Save data in json format

- Load data value instead of the whole object

- Inlcude the ID when listing people

CakeInTech committed:
decrease the complexity of the  method

christianbiring1 reviewed (approved):
Hi @MohamedCK  @Harrynoqs  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# STATUS: APPROVED :tada: 

## Highlights
- [x] The data can be saved correctly after the app is closed :+1: 
- [x] All linters checks are green :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
